Makefile

```diff
@@ -0,0 +1,290 @@
+.PHONY: help
+# target: help - Print this help
+help:
+	@egrep "^# target: " Makefile \
+		| sed -e 's/^# target: //g' \
+		| sort \
+		| awk '{printf("    %-20s", $$1); $$1=$$2=""; print "-" $$0}'
+
+.PHONY: build
+# target: build - Build package, run tests and create distribution
+build: .asdf check-epmd
+	@mvn
+
+.PHONY: clouseau1
+# target: clouseau1 - Start local inistance of clouseau1 node
+clouseau1: .asdf check-epmd
+	@mvn scala:run -Dname=$@
+
+.PHONY: clouseau2
+# target: clouseau2 - Start local inistance of clouseau2 node
+clouseau2: .asdf check-epmd
+	@mvn scala:run -Dname=$@
+
+.PHONY: clouseau3
+# target: clouseau3 - Start local inistance of clouseau3 node
+clouseau3: .asdf check-epmd
+	@mvn scala:run -Dname=$@
+
+.PHONY: clean
+# target: clean - Remove build artifacts
+clean:
+	@mvn clean
+
+check-epmd:
+	@if ! lsof -i TCP:4369 | grep -q epmd ; then echo "Error: Please start 'epmd' first"; exit 1; fi
```

Comment:
Is not `pgrep` ("find process by name") enough for that?  I might be wrong here, but `lsof` is usually an optional third-party software and its presence cannot be warranted.  On the contrary, `pgrep` is part of the base system.  Another option could be to not check for `empd` every time but launch it — if it is not running, it will be, otherwise that is a no-op because `empd` can detect itself and avoids running in multiple instances.

Reply:
I couldn't find information on what is more portable.

- lsof
- netstat
- pgrep

It seems lsof is available on macos and most of the linux by default. I am currently checking pgrep.

Reply:
I did check lsb-release and POSIX standards. They explicitly mention that networking tools are out of the scope. So don't know what is more portable. Keep trying...

Reply:
I can chain `ps` and `grep`.

Reply:
> I can chain `ps` and `grep`

The `ps` command behaves very differently between operating systems that follow BSD and AT&T semantics.

😢

Reply:
I'll use epmd itself

```
❯ epmd -names
epmd: Cannot connect to local epmd
```

Reply:
Addressed in e362d8792b4bd3bf3750f822c01694af94d2d8c0